components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2314,6 +2317,12 @@ do
   fi
 done
 
+# Test to ensure package name contains only valid characters
+if [[ ! "${pkg_name}" =~ ^[A-Za-z0-9_-]+$ ]];
```

Comment:
No doesn't seem to need it - it works correctly with a hypen in the package name.